Scripts/Runtime/Data/VirtualData/VDInstanceWrapper.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    internal readonly struct VDInstanceWrapper<T> : IEquatable<VDInstanceWrapper<T>>
+        where T : unmanaged, IEntityProxyData
+    {
+        public static bool operator==(VDInstanceWrapper<T> lhs, VDInstanceWrapper<T> rhs)
+        {
+            return lhs.ID == rhs.ID;
```

Comment:
No, we're making an optimization because we know how the `VDInstanceWrapper` will be used. 

It only exists to signify the origin from where data of type T came from. (Entity and Context). There can never be two instances of the same Data wrapped in any of the collections. 

I have added a comment in case someone dives into this and wonders though.